File: Invoicing.UI/src/services/customers/models/update-customer-request-model.ts
export interface UpdateCustomerRequestModel {
  name: string;
  street: string;
  houseNumber: string;
  postalCode: string;
  city: string;
  country: string;
  contactPerson: string | null;
  email: string | null;
  phoneNumber: string | null;
  vatNumber: string | null;
}

File: Invoicing.API/Invoicing.API/Services/Customers/Models/UpdateCustomerRequestModel.cs
using System.ComponentModel.DataAnnotations;

namespace Invoicing.API.Services.Customers.Models;

public class UpdateCustomerRequestModel
{
    [Required]
    public string Name { get; set; } = string.Empty;

    [Required]
    public string Street { get; set; } = string.Empty;

    [Required]
    public string HouseNumber { get; set; } = string.Empty;

    [Required]
    public string PostalCode { get; set; } = string.Empty;

    [Required]
    public string City { get; set; } = string.Empty;

    [Required]
    public string Country { get; set; } = "Nederland";

    public string? ContactPerson { get; set; }

    [EmailAddress]
    public string? Email { get; set; }

    public string? PhoneNumber { get; set; }
    public string? VatNumber { get; set; }
}
